internal/integrations/rawtemplate.go

```diff
@@ -0,0 +1,240 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package integrations // import "github.com/elastic/opentelemetry-collector-components/internal/integrations"
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/pipeline"
+)
+
+// RawTemplate implements the Template interface for raw YAML content.
+// Unused components are removed after resolving it, so the variables they contain are not required.
+type RawTemplate struct {
+	source rawYAMLConfig
+}
+
+// NewRawTemplate creates a new RawTemplate from raw YAML content.
+func NewRawTemplate(raw []byte) (*RawTemplate, error) {
+	var source rawYAMLConfig
+	decoder := yaml.NewDecoder(bytes.NewReader(raw))
+	decoder.KnownFields(true)
+	err := decoder.Decode(&source)
+	if err != nil {
+		return nil, fmt.Errorf("invalid integration template format: %w", err)
+	}
+	if err := source.Validate(); err != nil {
+		return nil, fmt.Errorf("integration template validation failed: %w", err)
+	}
+	return &RawTemplate{source: source}, nil
+}
+
+// Resolve resolves the template using a confmap resolver.
+func (t *RawTemplate) Resolve(ctx context.Context, params map[string]any, pipelines []pipeline.ID) (*Config, error) {
+	selectedPipelines := t.source.Pipelines
+	if len(pipelines) > 0 {
+		var err error
+		selectedPipelines, err = selectComponents(t.source.Pipelines, pipelines)
+		if err != nil {
+			return nil, fmt.Errorf("selecting pipelines: %w", err)
+		}
+	}
+	selectedReceivers, err := selectComponents(t.source.Receivers, listReceivers(selectedPipelines))
+	if err != nil {
+		return nil, fmt.Errorf("selecting receivers: %w", err)
+	}
+	selectedProcessors, err := selectComponents(t.source.Processors, listProcessors(selectedPipelines))
+	if err != nil {
+		return nil, fmt.Errorf("selecting processors: %w", err)
```

Comment:
No error of selectComponents is part of an API so that one could use it in any way.


```suggestion
		return nil, fmt.Errorf("selecting processors: %v", err)
```